src/compiler/scala/tools/nsc/backend/jvm/BCodeAsmCommon.scala

```diff
@@ -300,4 +302,73 @@ final class BCodeAsmCommon[G <: Global](val global: G) {
     }
     interfaces.map(_.typeSymbol)
   }
+
+  /**
+   * This is a hack to work around SI-9111. The completer of `methodSym` may report type errors. We
+   * cannot change the typer context of the completer at this point and make it silent: the context
+   * captured when creating the completer in the namer. However, we can temporarily replace
+   * global.reporter (it's a var) to store errors.
+   */
+  def completeSilentlyAndCheckErroneous(sym: Symbol): Boolean = {
+    if (sym.rawInfo.isComplete) false
+    else {
+      val originalReporter = global.reporter
+      val storeReporter = new reporters.StoreReporter()
+      try {
+        global.reporter = storeReporter
```

Comment:
Minor nitpick, but I would put this before the `try`. If it fails, there is nothing to rollback.